src/partition/repart.c

```diff
@@ -3172,6 +3358,130 @@ static int context_mkfs(Context *context) {
         return 0;
 }
 
+static int do_verity_format(
+                LoopDevice *data_device,
+                LoopDevice *hash_device,
+                uint64_t sector_size,
+                uint8_t **ret_roothash,
+                size_t *ret_roothash_size) {
+
+#if HAVE_LIBCRYPTSETUP
+        _cleanup_(sym_crypt_freep) struct crypt_device *cd = NULL;
+        _cleanup_free_ uint8_t *rh = NULL;
+        size_t rhs;
+        int r;
+
+        assert(data_device);
+        assert(hash_device);
+        assert(sector_size > 0);
+        assert(ret_roothash);
+        assert(ret_roothash_size);
+
+        r = dlopen_cryptsetup();
+        if (r < 0)
+                return log_error_errno(r, "libcryptsetup not found, cannot setup verity: %m");
+
+        r = sym_crypt_init(&cd, hash_device->node);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate libcryptsetup context: %m");
+
+        r = sym_crypt_format(
+                        cd, CRYPT_VERITY, NULL, NULL, NULL, NULL, 0,
+                        &(struct crypt_params_verity){
+                                .data_device = data_device->node,
+                                .flags = CRYPT_VERITY_CREATE_HASH,
+                                .hash_name = "sha256",
```

Comment:
doesn't need to be done now, but there are a lot of parameters that can be decided by the caller when creating the verity device, including the hash type - should at least be an item in the TODO to add parameters for hash type, FEC, etc